packages/cli/src/lib/snapshot/reportViewer.ts

```diff
@@ -0,0 +1,185 @@
+import {
+  ResourceSnapshotsReportModel,
+  ResourceSnapshotsReportOperationModel,
+  ResourceSnapshotsReportOperationResults,
+  ResourceSnapshotsReportResultCode,
+} from '@coveord/platform-client';
+import {cli} from 'cli-ux';
+import {bgHex, green, yellow, red, bold, italic} from 'chalk';
+
+export class ReportViewer {
+  private style = {
+    green: (txt: string) => green(txt),
+    yellow: (txt: string) => yellow(txt),
+    red: (txt: string) => red(txt),
+    header: (txt: string) => bold.hex('#1CEBCF')(txt),
+    error: (txt: string) => bgHex('#F64D64').hex('#272C3A')(txt),
+  };
+
+  public constructor(private readonly report: ResourceSnapshotsReportModel) {}
+
+  public display(): void {
+    this.printTable();
+
+    if (!this.isSuccessReport()) {
+      this.handleReportErrors();
+    }
+  }
+
+  private printTable() {
+    if (this.changedResources.length === 0) {
+      cli.log(this.style.header('\nNo changes detected'));
+      return;
+    }
+
+    cli.table(this.changedResources, {
+      resourceName: {
+        header: this.style.header('\nPreviewing resource changes:'),
+        get: (row) => this.printTableSection(row),
+      },
+    });
+  }
+
+  // TODO: Change logic once SRC-4448 is complete
+  private printTableSection(row: {
+    resourceName: string;
+    operations: ResourceSnapshotsReportOperationModel;
+  }) {
+    const resourceType = this.prettyPrintResourceName(row.resourceName);
```

Comment:
This method works, but I think it's going to be somewhat tricky to change/evolve over time, because it relies on precise string alignment baked into each string output. And could be easy to end up with things un-aligned in the console when we do a change.

One potential refactor would be to create a class for `PreviewDisplay` (or PreviewFormatter).

Something like this (very crude pseudo code):
```

class PreviewDisplayRow() {
     contructor(text, type) {
        // where type is one of success/warning/error, etc.
     }

    public render() {
       // return string color formatted     
    }
}


class PreviewDisplay() {
   private rows: PreviewDisplayRow[];

   public addRow(text, type) {
         this.rows.push(new PreviewDisplayRow(text, type);
         return this;
    }

   render() {
       // call render on each row, and format them on individual line
   }
}
```

Do you think it would be helpful for the more "advanced" preview if we already have this created ?
Or that it would needlessly complicate things ?

Reply:
@olamothe I get your point, but I don't think we need to worry about the alignment as it is already taken care of by the `cli-ux` package. The `cli.table` util will compute the length of the longest row and will apply the necessary adjustments. It also supports multi-line rows (as we have here).